dashboard/app/controllers/api/v1/sections_controller.rb

```diff
@@ -47,7 +47,7 @@ def create
         login_type: params[:login_type],
         grade: Section.valid_grade?(params[:grade].to_s) ? params[:grade].to_s : nil,
         script_id: script_to_assign ? script_to_assign.id : params[:script_id],
-        course_id: params[:course_id] && UnitGroup.valid_course_id?(params[:course_id]) ?
+        course_id: params[:course_id] && UnitGroup.valid_course_id?(params[:course_id], current_user) ?
```

Comment:
I don't want to block this PR (so feel free to merge!) but this feels like really odd behavior in general. The script id is set correctly but the course isn't so does that mean a non-pilot teacher could have a pilot script assigned to their section, in theory?

Reply:
I agree. This is a preexisting issue so I'll start a thread on it to get the right team ownership

Reply:
It is functionally unlikely this would happen in the real world because we don't show pilot scripts to non-pilot teachers but I think it's still worth cleaning up.